.github/workflows/canary.yml

```diff
@@ -11,8 +11,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
-      - uses: actions/setup-go@v2
+        uses: actions/checkout@ee0669bd1cc54295c223e0bb666b733df41de1c5 # ratchet:actions/checkout@v2
```

Comment:
Why do we pin to this GHA?

Reply:
It's a vector for supply chain attacks, eg. [this](https://snyk.io/blog/reconstructing-tj-actions-changed-files-github-actions-compromise/) recent one. Version tags are not immutable, so if attackers can compromise the action repository, they can redirect existing tags to a compromised version of the action.

Reply:
Thank you for the explanation!